internal/integration/integration_test.go

```diff
@@ -128,17 +139,96 @@ func (suite *IntegrationSuite) TearDownSuite() {
 
 // TestScaleControlPlane scale control plane nodes.
 func (suite *IntegrationSuite) TestScaleControlPlane() {
-	err := suite.cluster.Scale(suite.ctx, 3, capi.ControlPlaneNodes)
+	selector, err := labels.Parse("cluster.x-k8s.io/control-plane")
 	suite.Require().NoError(err)
 
-	time.Sleep(2 * time.Second)
+	suite.cluster.Scale(suite.ctx, 3, capi.ControlPlaneNodes) //nolint:errcheck
 
-	suite.Require().NoError(suite.cluster.Sync(suite.ctx))
+	suite.Require().NoError(suite.cluster.Health(suite.ctx))
+
+	client, err := suite.manager.GetClient(suite.ctx)
+	suite.Require().NoError(err)
+
+	machines := unstructured.UnstructuredList{}
+	machines.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "cluster.x-k8s.io",
+		Kind:    "Machine",
+		Version: suite.manager.Version(),
+	})
+
+	suite.Require().NoError(client.List(suite.ctx, &machines, &runtimeclient.MatchingLabelsSelector{
+		Selector: selector,
+	}))
 
-	if talosVersion.LessThan(*scaleVersion) {
-		suite.T().Skip("skip for Talos <= v0.12.2, scale down is unstable")
+	getEtcdMembers := func() (map[uint64]struct{}, error) {
+		members := map[uint64]struct{}{}
+
+		talosclient, e := suite.cluster.TalosClient(suite.ctx)
+		if e != nil {
+			return nil, e
+		}
+
+		resp, e := talosclient.EtcdMemberList(suite.ctx, &machineapi.EtcdMemberListRequest{})
+		if e != nil {
+			return nil, e
+		}
+
+		for _, m := range resp.Messages[0].Members {
+			members[m.Id] = struct{}{}
+		}
+
+		return members, nil
 	}
 
+	oldEtcdMembers, err := getEtcdMembers()
+	suite.Require().NoError(err)
+
+	suite.Require().Greater(len(machines.Items), 0)
+
+	machine := machines.Items[0]
+
+	suite.Require().NoError(client.Delete(suite.ctx, &machine))
```

Comment:
should we put this as a separate test? as this is something close to MachineDeploymentReconcile in sfyra/sidero: drop a machine, wait for health; this is not using scale path, but rather relying on CAPI recreating the missing machine (and CACCPT fixing the etcd state).

Reply:
Initially I did it like that, but downside of that approach is that you'll have to scale up before test and then scale everything down after it (if we want tests to be idempotent) and as scaling up and down is super slow this test will add another 10-15 minutes to the whole test duration :(

Reply:
I don't know if it's dumb idea or not really, but we could split the tests and use numbers or something to order them properly (iirc, go test runs them in alphabetical order?), so that we could do `Test01ScaleUp`, `Test02ReconcileMachine`, `Test03ScaleDown`. 

this is close to what we do in Sfyra - control the sequence of tests

Reply:
I was trying to avoid making them order dependent. But that works, yes.